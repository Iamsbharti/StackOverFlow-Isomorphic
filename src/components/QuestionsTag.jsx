import React from "react";
import { Link } from "react-router-dom";
function QuestionsTag({ tags }) {
  return (
    <div>
      {tags.map((tag) => (
        <Link to={`/tags/${tag}`} key={tag}>
          <code key={tag}>{tag},</code>
        </Link>
      ))}
    </div>
  );
}
export default QuestionsTag;
