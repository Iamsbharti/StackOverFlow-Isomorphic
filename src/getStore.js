import { createStore, combineReducers, applyMiddleware } from "redux";
import { identity } from "lodash";
import createSagaMiddleWare from "redux-saga";
import { createLogger } from "redux-logger";
import fetchQuestionsSaga from "./saga/get-questions-saga";
import fetchQuestionSaga from "./saga/get-question-saga";
import fetchTaggedQuestions from "./saga/get-tag-que-saga";
import * as reducers from "./reducers";
import { routerReducer as router, routerMiddleware } from "react-router-redux";
export default function getStore(history, defaultState) {
  //create a middle ware for saga
  const sagaMiddleWare = createSagaMiddleWare();
  //middleware to handle routes
  const middleWare = routerMiddleware(history);
  //a chain for all the middlewares
  const middleWareChain = [middleWare, sagaMiddleWare];
  //a logger is create only for development envoirnment
  if (process.env.NODE_ENV === "development") {
    const logger = createLogger();
    middleWareChain.push(logger);
  }

  //create a store and apply the middleware components
  console.log("creating store");
  const store = createStore(
    combineReducers({ ...reducers, router }),
    defaultState,
    applyMiddleware(...middleWareChain)
  );
  //run saga
  sagaMiddleWare.run(fetchQuestionsSaga);
  sagaMiddleWare.run(fetchQuestionSaga);
  sagaMiddleWare.run(fetchTaggedQuestions);
  return store;
}
