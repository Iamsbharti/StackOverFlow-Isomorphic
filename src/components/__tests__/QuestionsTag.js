import React from "react";
import renderer from "react-test-renderer";
import QuestionsTag from "../QuestionsTag";
import getStore from "../../getStore";
import { Provider } from "react-redux";
import { ConnectedRouter } from "react-router-redux";
import createHistory from "history/createBrowserHistory";

//create history for connected router and pass it to store
const history = createHistory();
describe("Test QuestionTag component using snapshot-test", () => {
  it("should render correctly and not regress the QuestionTag Component", () => {
    const questionTagTree = renderer.create(
      <Provider store={getStore()}>
        <ConnectedRouter history={history}>
          <QuestionsTag tags={["react", "html"]} />
        </ConnectedRouter>
      </Provider>
    );
    expect(questionTagTree.toJSON()).toMatchSnapshot();
  });
});
