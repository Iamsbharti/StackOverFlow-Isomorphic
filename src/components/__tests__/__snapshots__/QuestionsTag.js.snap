// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QuestionTag component using snapshot-test should render correctly and not regress the QuestionTag Component 1`] = `
<div>
  <a
    href="/tags/react"
    onClick={[Function]}
  >
    <code>
      react
      ,
    </code>
  </a>
  <a
    href="/tags/html"
    onClick={[Function]}
  >
    <code>
      html
      ,
    </code>
  </a>
</div>
`;
